src/terminal/adapter/adaptDispatchGraphics.cpp

```diff
@@ -259,6 +259,38 @@ bool AdaptDispatch::SetGraphicsRendition(const VTParameters options)
     return true;
 }
 
+// Routine Description:
+// - DECSCA - Modifies the character protection attribute. This operation was
+//   originally intended to support a range of logical character attributes,
+//   but the protected attribute was the only one ever implemented.
+// Arguments:
+// - options - An array of options that will be applied in order.
+// Return Value:
+// - True.
+bool AdaptDispatch::SetCharacterProtectionAttribute(const VTParameters options)
+{
+    auto& textBuffer = _api.GetTextBuffer();
+    auto attr = textBuffer.GetCurrentAttributes();
+    for (size_t i = 0; i < options.size(); i++)
+    {
+        const LogicalAttributeOptions opt = options.at(i);
+        switch (opt)
+        {
+        case Default:
+            attr.SetProtected(false);
+            break;
+        case Protected:
+            attr.SetProtected(true);
+            break;
+        case Unprotected:
+            attr.SetProtected(false);
+            break;
+        }

```

Comment:
But what about this?
```cpp
switch (opt)
{
case Default:
case Unprotected: // <-- deduplicated
    attr.SetProtected(false);
    break;
case Protected:
    attr.SetProtected(true);
    break;
default: // <-- explicit (maybe?)
    break;
}
```

Reply:
I did consider that, but the reason I left the `Default` separate from the `Unprotected` was because semantically they're actually different. `Default` serves the same purpose as `SGR 0` does for rendition attributes - it resets _all_ of the logical attributes. For now that just means resetting the protected attribute, because that's all there is, but if we ever added others, those two cases would definitely be different. I thought it better to get the structure right up front rather than trying to over optimise it.